fendermint/testing/materializer/tests/docker_tests/benches.rs

```diff
@@ -0,0 +1,494 @@
+    chain_id: U256,
+) -> anyhow::Result<TestMiddleware<C>>
+where
+    C: JsonRpcClient,
+{
+    let wallet: Wallet<SigningKey> = Wallet::from_bytes(sender.secret_key().serialize().as_ref())?
+        .with_chain_id(chain_id.as_u64());
+
+    Ok(SignerMiddleware::new(provider, wallet))
+}
+
+#[serial_test::serial]
+#[tokio::test]
+async fn test_native_coin_transfer() -> Result<(), anyhow::Error> {
+    let (testnet, cleanup) = make_testnet(MANIFEST, |_| {}).await?;
+
+    let block_gas_limit = U256::from(10_000_000_000u64);
+    let max_tx_gas_limit = U256::from(3_000_000u64);
+
+    let pangea = testnet.node(&testnet.root().node("pangea"))?;
+    let provider = pangea
+        .ethapi_http_provider()?
+        .expect("ethapi should be enabled");
+    let chain_id = provider
+        .get_chainid()
+        .await
+        .context("failed to get chain ID")?;
+
+    let cancel = Arc::new(Signal::new());
+
+    // Set up background blocks collector.
+    let blocks_collector = {
+        let cancel = cancel.clone();
+        let provider = provider.clone();
+        let assert = move |block: &Block<H256>| {
+            // Make sure block gas limit isn't the bottleneck.
+            let unused_gas_limit = block_gas_limit - block.gas_limit;
+            assert!(unused_gas_limit >= max_tx_gas_limit);
+        };
+        tokio::spawn(collect_blocks(cancel, provider, assert))
+    };
+
+    // Drive concurrency.
+    let cfg = Execution::new()
+        .add_step(1, 5)
+        .add_step(10, 5)
+        .add_step(100, 5)
+        .add_step(150, 5);
+    let testnet = Arc::new(testnet);
+    let testnet_clone = testnet.clone();
+    let nonce_manager = Arc::new(NonceManager::new());
+
+    let results = concurrency::execute(cfg.clone(), move |mut input| {
+        let testnet = testnet_clone.clone();
+        let nonce_manager = nonce_manager.clone();
+        let provider = provider.clone();
+
+        let test = async move {
+            let sender = testnet.account_mod_nth(input.test_id);
+            let recipient = testnet.account_mod_nth(input.test_id + 1);
+            println!("running (test_id={})", input.test_id);
+
+            let middleware = make_middleware(provider, sender, chain_id)
+                .await
+                .context("make_middleware")?;
+
+            let sender: H160 = sender.eth_addr().into();
+            let nonce = nonce_manager.get_and_increment(sender).await;
+
+            // Create the simplest transaction possible: send tokens between accounts.
+            let to: H160 = recipient.eth_addr().into();
+            let mut tx = Eip1559TransactionRequest::new()
+                .to(to)
+                .value(1)
+                .nonce(nonce);
+
+            let gas_estimation = middleware
+                .estimate_gas(&tx.clone().into(), None)
+                .await
+                .unwrap();
+            tx = tx.gas(gas_estimation);
+            assert!(gas_estimation <= max_tx_gas_limit);
```

Comment:
Should this fail to whole test run?

Reply:
I don't see a reason for allowing this to pass, and having to deal with partial failures. However, this shouldn't normally fail, and if so should fail consistently.